hooks/mari/collector.py

```diff
@@ -207,12 +207,11 @@ def collect_texture_elements(self, settings, parent_item):
                 self.logger.info("Collected item: %s" % channel_item.name)
                 items.append(channel_item)
 
-                if len(collected_layers) > 0 and layers_item is None:
```

Comment:
`len(collected_layers) > 0` is redundant as we have an early exit in lines 187-189 (channel itself is not exported if there are no layers)
`layers_item` is initialised for the first channel and becomes not None. This becomes False for the subsequent channels and so all layers are appended to the layers_item parented under the first channel_item. I have no idea why this check was put in place at all.

Reply:
Do we need to publish layers as well if we have exactly 1 layer? That and flattened channel would be the same in this case.